technical-documents/postmortems/2022-02-02-inflation-vuln.md

```diff
@@ -0,0 +1,157 @@
+# Self-Destruct Inflation Vulnerability Postmortem
+
+This document describes a critical bug in our system which we were alerted to on February 2nd, 2022.
+It also details our response, lessons learned, and subsequent changes to our processes.
+
+## Incident Summary
+
+A vulnerability in Optimism’s fork of go-ethereum ([L2-geth](../../l2geth/README.md)) was reported to us by [Jay Freeman](https://twitter.com/saurik) (AKA saurik) on February 2nd, 2022. If exploited, this vulnerability would allow anyone to mint an unbounded amount of ETH on Optimism.
+
+We confirmed the issue, implemented a fix, and upgraded our infrastructure within 3 hours. We then alerted infrastructure providers running Optimism, as well as other network operators who are using a fork of our software.
+
+All parties were running a patched version of L2-geth within 32 hours of the initial report.
+
+## Lead up
+
+saurik had been engaging with our code, and [opening issues](https://github.com/ethereum-optimism/optimism/issues?q=is%3Aissue+author%3Asaurik+) for several months prior to identifying this bug.
+
+We launched our [Immunefi-hosted bug bounty program](https://immunefi.com/bounty/optimism/) on January 13th, 2022, slightly more than 2 weeks before his report. The maximum payout for the program was $2,000,042. According to saurik, his decision to hunt for bugs in our code had two motivations. One was the financial reward, the other was needing a topic to speak about at the upcoming ETHDenver conference.
+
+## The Vulnerability
+
+Contract balances were improperly zeroed during self-destruction, so that the contract address would still have a balance after it had been self-destructed. This could have enabled an attacker to run a loop which doubled the balance of a contract each time, resulting in massive inflation and issuance directly to the attacker.
+
+A thorough description can be found in saurik's [writeup](https://www.saurik.com/optimism.html).
+
+## Impact
+
+The issue was not exploited, so there was no impact to ordinary users. However, the issue required node operators to update as quickly as possible. Infrastructure providers were especially impacted, since they had to roll out an emergency patch version.
+
+## Detection
+
+Jay Freeman (a.k.a. Saurik) reported the bug to us via security@optimism.io. He first attempted to report via our Immunefi bounty program, but decided to email us since it does not explicitly include our L2-Geth code.
+
+## Response
+
+- @Kelvin Fichter, @Mark Tyneway, @Karl Floersch, and @protolambda responded to the report and reproduced the issue.
```

Comment:
Prefer to use my github handle instead of my name (@smartcontracts)

Reply:
Done.